### PR TITLE
MCOL-128 Support ALTER TABLE...ENGINE=Columnstore

### PR DESCRIPTION
**Pull request by LinuxJedi**
Also implements:

* ALTER TABLE from Columnstore to another engine
* MCOL-3349 CREATE TABLE ... AS SELECT.

### CODE REVIEW

File: dbcon/mysql/ha_mcs_ddl.cpp [hunk 1]
```diff
@@ -2600,7 +2607,7 @@ int ha_mcs_impl_delete_table_(const char* db, const char* name, cal_connection_i
 
     string emsg;
 
-    if (thd->lex->sql_command == SQLCOM_DROP_DB)
+    if ((thd->lex->sql_command == SQLCOM_DROP_DB) || (thd->lex->sql_command == SQLCOM_ALTER_TABLE))
```
Comment (pleblanc1976) [inline dbcon/mysql/ha_mcs_ddl.cpp:2610]: This is for the 'alter table .. engine = non-CS-engine' case I assume?

**Review by pleblanc1976 — CHANGES_REQUESTED** [review]
Buildbot had a problem in test000.  I built locally to see if it was a BB problem or a code problem.  Looks like a code problem.

server commit = 8a931e4d16 (current 10.4 branch of server repo)
engine commit = d0a910cbdd6 (commit BB checked out)

Getting this running simple DDL:
`MariaDB [tpch1]> create table test (i int) engine=columnstore;
ERROR 1178 (42000): The storage engine for the table doesn't support IDB-4016: DML and DDL statements for Columnstore tables can only be run from the replication master.
MariaDB [tpch1]>`

File: dbcon/mysql/ha_mcs_ddl.cpp (hunk 1 above)
Comment (LinuxJedi) [reply dbcon/mysql/ha_mcs_ddl.cpp:2610]: Correct, the final phase is to drop the table in the old engine. If you don't do this it succeeds but generates an error and then you are in a weird state where the table is in both ColumnStore and the other engine.

**Comment by LinuxJedi** [comment]
Weird, I didn't touch that code and test000 passed locally. I'll look into it.

**Review by pleblanc1976 — APPROVED** [review]
C7 BB is happy now, so will merge.

**Pull request merged**